PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx

```diff
@@ -344,7 +345,10 @@ struct HfCorrelatorDplusHadrons {
         for (unsigned int iclass = 0; iclass < classMl->size(); iclass++) {
           outputMl[iclass] = candidate.mlProbDplusToPiKPi()[classMl->at(iclass)];
         }
-        entryDplusCandRecoInfo(hfHelper.invMassDplusToPiKPi(candidate), candidate.pt(), outputMl[0], outputMl[1]); // 0: BkgBDTScore, 1:PromptBDTScore
+        if (isPromptAnalysis)
+          entryDplusCandRecoInfo(hfHelper.invMassDplusToPiKPi(candidate), candidate.pt(), outputMl[0], outputMl[1]); // 0: BkgBDTScore, 1:PromptBDTScore
+        else
+          entryDplusCandRecoInfo(hfHelper.invMassDplusToPiKPi(candidate), candidate.pt(), outputMl[0], outputMl[2]); // 0: BkgBDTScore, 2:FDBDTScore
```

Comment:
The `if else` blocks are unnecessary duplication of code, since the only difference between the two cases is the index of `outputMl`. Instead, define a member variable for the index value and set it in `init` based on the value of `isPromptAnalysis`.

Reply:
Hi Vit, Thanks. I implemented an integer index, which is initialized once in the init() function and then used in all the processes.